IdentityCore/src/broker_operation/request/MSIDBrokerOperationRequest.m

```diff
@@ -106,9 +125,23 @@ - (NSDictionary *)jsonDictionary
     return json;
 }
 
--(NSString *)logInfo
+- (NSString *)logInfo
 {
     return [NSString stringWithFormat:@"%@",[self.jsonDictionary msidMaskedRequestDictionary]];
 }
 
++ (BOOL)shouldIgnoreBrokerKey
+{
+#if !TARGET_OS_OSX
+    return NO;
+#else
+        if([MSIDKeychainUtil sharedInstance].isAppEntitled)
```

Comment:
Should we skip it only for MSAL runtime scenarios?

Reply:
Even if that's the case, we should still put it behind some sort of a preprocessing macro. My concern is that this enables MSAL developers to be not properly signed/entitled, which is not something we would want to do.

Reply:
Got it, there should be a flag and I'll check how to plugging to our code

Reply:
Updated by adding a new property